librz/core/cbin.c

```diff
@@ -3082,16 +3141,38 @@ static int bin_sections(RzCore *r, PJ *pj, int mode, ut64 laddr, int va, ut64 at
 				: section->name;
 			// seems like asm.bits is a bitmask that seems to be always 32,64
 			// const char *asmbits = rz_str_sysbits (bits);
-			if (hashtypes) {
-				rz_table_add_rowf(table, "dXxXxsss", i,
-					(ut64)section->paddr, (ut64)section->size,
-					(ut64)addr, (ut64)section->vsize,
-					perms, hashstr, section_name);
+			if (!print_segments && plugin_type_flags_support) {
```

Comment:
also here, to avoid duplication, I would construct the table row incrementally and then just do an add at the end. `i, (ut64)section->paddr, (ut64)section->size, (ut64)addr, (ut64)section->vsize,`are common to all sections, for example. You could create a RzList and append the values for each column similar to how i proposed above with columns setup. Then you can use `rz_table_add_row_list`.